src/gennylib/include/gennylib/conventions.hpp

```diff
@@ -124,12 +127,91 @@ inline bool operator==(const RateSpec& lhs, const RateSpec& rhs) {
     return (lhs.per == rhs.per) && (lhs.operations == rhs.operations);
 }
 
+struct PhaseRangeSpec {
+    PhaseRangeSpec() = default;
+    ~PhaseRangeSpec() = default;
+
+    PhaseRangeSpec(genny::IntegerSpec s, genny::IntegerSpec e) {
+        assert(s.value >= 0 && e.value >= 0 && s.value <= UINT_MAX && e.value <= UINT_MAX && s.value <= e.value);
+        start = static_cast<genny::PhaseNumber>(s.value);
+        end = static_cast<genny::PhaseNumber>(e.value);
+    }
+    PhaseRangeSpec(genny::IntegerSpec s) : PhaseRangeSpec(s, s) {}
+
+    // Allow construction with unsigned ints for testing
+    PhaseRangeSpec(u_int32_t s, u_int32_t e) : start{s}, end{e} {}
+    PhaseRangeSpec(u_int32_t s) : PhaseRangeSpec(s, s) {}
+
+    genny::PhaseNumber start;
+    genny::PhaseNumber end;
+};
+
 }  // namespace genny
 
 namespace YAML {
 
 using genny::decodeNodeInto;
 
+/**
+ * Convert between YAML and genny::PhaseRange
+ *
+ * The YAML syntax accepts "[genny::Integer]..[genny::Integer]".
+ * This is used to stipulate repeating a phase N number of times
+ */
+template <>
+struct convert<genny::PhaseRangeSpec> {
+    static Node encode(const genny::PhaseRangeSpec rhs) {
+        std::stringstream msg;
+        msg << rhs.start << ".." << rhs.end;
+        return Node{msg.str()};
+    }
+
+    static bool decode(const Node& node, genny::PhaseRangeSpec& rhs) {
+        if (node.IsSequence() || node.IsMap()) {
+            return false;
+        }
+        auto strRepr = node.as<std::string>();
+
+        // use '..' as delimiter.
+        const std::string delimiter = "..";
+        auto delimPos = strRepr.find(delimiter);
+
+        if (delimPos == std::string::npos) {
+            // check if user inputted an integer
+            try {
+                auto phaseNumberYaml = node.as<genny::IntegerSpec>();
+                rhs = genny::PhaseRangeSpec(phaseNumberYaml);
+                return true;
+            } catch (const genny::InvalidConfigurationException) {
+                std::stringstream msg;
+                msg << "Invalid value for genny::PhaseRangeSpec: '" << strRepr << "'."
+                    << " The correct syntax is either a single integer or two integers delimited "
+                       "by '..'";
+                throw genny::InvalidConfigurationException(msg.str());
+            }
+        }
+
+        genny::IntegerSpec start;
+        genny::IntegerSpec end;
+
+        auto startYaml = Load(strRepr.substr(0, delimPos));
+        auto endYaml = Load(strRepr.substr(delimPos + delimiter.size()));
```

Comment:
it would be nice to be able to space-separate them e.g.

```yaml
Phase: 1 .. 5
```

I think if you do [`atoi`](http://www.cplusplus.com/reference/cstdlib/atoi/) to both sides of the `..` it'll correctly strip leading/trailing spaces. But it may also mask invalid input since I think `atoi("123abc") == 123`. Your call if you think this is worth it or not. Either way please add a test case for what happens if you specify `Phase: 1 .. 3`.

Reply:
Added a test for this. Parsing the two sides of the `..` into an `IntegerSpec` handles this for free which is nice.